### DISCUSSION TITLE
Infractions/Bans

### DISCUSSION THREAD

**Discussion by DoraKlikOpDora** [NamelessMC/Nameless#2123] [Ideas]
# I do not like the way infractions is included in Nameless, because...

Infractions and Unban appeal are literally not working together. The idea behind banning a player and having linked a bansystem, like BAT, is that you ban the player from the forums and from the game, but that the player is still able to create an unbanappeal.

At the moment you are banning a player from the website and it cannot login, but it does nothing with BAT. Ingame the player can still login. I think a solution for this is that banning isn't preventing from logging in, but preventing from sending direct messages and posting on the forum. A player who is banned should be able to login to their account, should be able to view and read everything, but should not be able to communicate and interact with other members of your community.

Why are we using the unbanappeal system while we can also use the forum? Okay, there are reasons, but mainly you are using it because you only want a player to communicate with the staff about their ban. Now you can't ban somebody of the website, because they can never appeal.

# **What has to happen**
* A banned player cannot visit pages with forms, except the unbanappeal page.
* Integration of bansystems has to be 100%. You must be able to perform actions on the website that will also be performed ingame: Kicking (where kicking is only ingame), banning (**but also unban**), warning, muting (where muting is a ban on the website and a mute ingame). A warning made on the website should pop-up as well as on the website as ingame.

Possible solution is creating a NamelessBAN plugin which is fully compatible with the website.

**Comment by Sorro** [comment]
There is a NamelessPlugin that is under DEV.
If you have any other suggestions for the plugin please post here.
https://github.com/NamelessMC/Nameless-Plugin/issues
Thx!

**Comment by MuhsinunC** [comment]
I created an addon that does JUST what ur second bulletpoint asks! :D I took the BAT Web Interface and then themed it to look like NamelessMC :D I submitted a pull request and I'm currently waiting on them to accept it :D
